GEECS-PythonAPI/geecs_python_api/analysis/scans/scan_data.py

```diff
@@ -38,68 +42,55 @@ def __init__(self, folder: Optional[SysPath] = None,
         experiment : str
               Experiment name, e.g. 'Undulator'.  Necessary if just given a tag
         """
-
-        self.identified = False
         self.scan_info: dict[str, str] = {}
 
         self.__folder: Optional[Path] = None
         self.__tag: Optional[ScanTag] = None
         self.__tag_date: Optional[date] = None
         self.__analysis_folder: Optional[Path] = None
 
+        self.data_dict = {}
+        self.data_frame = None  # use tdms.geecs_tdms_dict_to_panda
+
         if folder is None:
             if tag and experiment:
                 folder = self.build_scan_folder_path(tag, base_directory=base_path, experiment=experiment)
 
         if folder:
-            try:
-                folder = Path(folder)
-
-                (exp_name, year_folder_name, month_folder_name, date_folder_name,
-                 scans_literal, scan_folder_name) = folder.parts[-6:]
-
-                if (not re.match(r"Y\d{4}", year_folder_name)) or \
-                   (not re.match(r"\d{2}-(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)", month_folder_name)) or \
-                   (not re.match(r"\d{2}_\d{4}", date_folder_name)) or \
-                   (not scans_literal == 'scans') or \
-                   (not re.match(r"Scan\d{3,}", scan_folder_name)):
-                    raise ValueError("Folder path does not appear to follow convention")
-                elif not folder.exists():
-                    raise ValueError("Folder does not exist")
+            folder = Path(folder)
 
-                self.__tag_date = dtime.strptime(date_folder_name, "%y_%m%d").date()
-                self.__tag = \
-                    ScanTag(self.__tag_date.year, self.__tag_date.month, self.__tag_date.day, int(scan_folder_name[4:]))
+            (exp_name, year_folder_name, month_folder_name, date_folder_name,
+             scans_literal, scan_folder_name) = folder.parts[-6:]
 
```

Comment:
(exp_name, year_folder_name, month_folder_name, date_folder_name,
             scans_literal, scan_folder_name) = folder.parts[-6:]

I think this part should come after the checking if the folder follows expected formatting (lines 71+)

Reply:
This is more of an issue with how Github decided to present the changes.  I removed a try-except block so everything got un-indented, and it got very confused as to how to present the changes.  But if you look at the line numbers, the parsing is on lines 62-63 and the if statement is on lines 71-76